Sources/ViewControllers/Products/Detail/ProductAttributes/ProductDetailViewControllerExtension.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  ProductDetailViewControllerExtension.swift
+//  OpenFoodFacts
+//
+//  Created by Alexander Scott Beaty on 11/3/20.
+//
+import FloatingPanel
+
+extension ProductDetailViewController {
+    // MARK: - ProductAttributes FloatingPanel setup
+    func configureFloatingPanel() {
+
+        floatingPanelController = FloatingPanelController()
+        floatingPanelController.delegate = self
+        floatingPanelController.contentMode = .fitToBounds
+
+        // Add the floating panel view to the controller's view on top of other views.
+        self.view.addSubview(floatingPanelController.view)
+        floatingPanelController.view.frame = self.view.bounds
+
+        // In addition, Auto Layout constraints are highly recommended.
+        // Constraint the fpc.view to all four edges of your controller's view.
+        // It makes the layout more robust on trait collection change.
+        floatingPanelController.view.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+          floatingPanelController.view.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 0.0),
+          floatingPanelController.view.leftAnchor.constraint(equalTo: self.view.leftAnchor, constant: 0.0),
+          floatingPanelController.view.rightAnchor.constraint(equalTo: self.view.rightAnchor, constant: 0.0),
+          floatingPanelController.view.bottomAnchor.constraint(equalTo: self.view.bottomAnchor, constant: 0.0)
+        ])
+
+        let storyboard = UIStoryboard(name: "ProductAttributeViewController", bundle: nil)
+        // swiftlint:disable:next force_cast
+        productAttributeController = (storyboard.instantiateViewController(withIdentifier: "ProductAttributeViewController") as! ProductAttributeViewController)
+        floatingPanelController.set(contentViewController: productAttributeController)
+
+        floatingPanelController.surfaceView.backgroundColor = .clear
+        floatingPanelController.surfaceView.cornerRadius = 9.0
+        floatingPanelController.surfaceView.shadowHidden = false
+        // Add a gesture to hide the floating panel
+        let gestureDown = UISwipeGestureRecognizer(target: self, action: #selector(self.hideSummaryView(_:)))
+        gestureDown.numberOfTouchesRequired = 1
+        gestureDown.direction = .down
+        floatingPanelController.surfaceView.addGestureRecognizer(gestureDown)
+        floatingPanelController.surfaceView.isUserInteractionEnabled = true
+
+        floatingPanelController.addPanel(toParent: self)
+    }
+
+    // MARK: - Gesture recognizers
+    @objc func hideSummaryView(_ sender: UISwipeGestureRecognizer) {
+        floatingPanelController.move(to: FloatingPanelPosition.hidden, animated: true)
+    }
```

Comment:
@philippeauriach just wanted to confirm, was this the part you noticed? This code comes from ScannerViewController.swift with nothing changed I think. 

I plan on using Xcode Instruments: Allocations soon to make sure I am not introducing memory leaks, which I found tends to happen when the UI elements are not being destroyed.

Reply:
Yes this is the "swipe down" gesture I refered to, if it comes from the Scanner then there is no problem